prime-router/docs/design/validation/design.md

```diff
@@ -0,0 +1,271 @@
+# Problem Statement
+As a sender I want submitted data to be validated to ensure data integrity and quality. As a receiver I want received
+data to be validated to ensure data integrity and quality. For this to happen, there needs to exist a validation 
+setting as well as reusable validation components to perform the sending and receiving validation. 
+
+## Validation Design
+The diagram below proposes four validation "checkpoints":
+
+| Name           | Description + Location                                              | Purpose                                                                                                        |
+|----------------|---------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------|
+| Submission     | Upon receipt of a submission in the SenderReceiver                  | Ensure data is parseable (i.e. not a png or other garbage data)                                                |
+| Post-transform | After sender transforms and enrichment in the convert function      | Ensure data is valid after transformation (i.e. satisfies the validation profile configured for this sender)   | 
+| Pre-dispatch   | After receiver translation and enrichment in the translate function | Ensure data is valid after transformation (i.e. satisfies the validation profile configured for this receiver) | 
+
+This is where they are located in the overall system:  
+![annotated-architecture-diagram.png](annotated-architecture-diagram.png)
+
+### Validation Checkpoints
+
+#### Submission
+For FHIR, we currently use an `IParser` instance to parse and read some metadata. By default, the IParser performs some
+validation and raises warnings or errors (depending on how its configured). We need to investigate how the instance is
+currently configured and ensure it is performing validation. Ideally this configuration will be universal or binary
+(validation on/off vs different types) for all senders. For HL7 V2, we will use the HAPI library as it is currently 
+being used in the Covid pipeline
+
+#### Post transform
+Will occur after transformation so that we can massage the data to meet any IG that may be requested. We again need to 
```

Comment:
that may be requested `and do content validation for free via the HAPI library authentication as opposed to just structural.`